dynacred/src/personhood-rpc.ts

```diff
@@ -76,69 +79,89 @@ export class PersonhoodRPC {
 
     // wipeMeetups removes all meetups from the servers. This is mainly for tests.
     async wipeMeetups() {
-        return this.meetups(new Meetup(null, true));
+        return this.meetups(new Meetup({wipe: true}));
     }
 
-    async listRPS(id: InstanceID = null): Promise<RoPaSci[]> {
-        const ropasci: {newropasci: any} = {newropasci: null};
-        if (id) {
-            ropasci.newropasci = new RoPaSci(this.rpc.genesisID, id).toObject();
-        }
+    async listRPS(newRoPaSci: RoPaSci = null): Promise<RoPaSci[]> {
         const ropascis: RoPaSci[] = [];
+        let rpsList = new RoPaSciList();
+        if (newRoPaSci) {
+            rpsList = new RoPaSciList({newRoPaSci});
+        }
         await Promise.all(this.list.map(async (addr) => {
-            const socket = new WebSocket(addr.getWebSocketAddress(), PersonhoodRPC.serviceID);
-            // let resp = await socket.send("RoPaSciList", "RoPaSciListResponse", ropasci);
-            // if (resp && resp.ropascis) {
-            //     ropascis = ropascis.concat(resp.ropascis.map(r => RoPaSci.fromObject(r)));
-            // }
+            const socket = new WebSocketConnection(addr.getWebSocketAddress(), PersonhoodRPC.serviceID);
+            const resp: RoPaSciListResponse = await socket.send(rpsList, RoPaSciListResponse);
+            if (resp && resp.roPaScis) {
+                ropascis.push(...resp.roPaScis);
+            }
         }));
         return ropascis.filter((rps, i) => {
-            return ropascis.findIndex((p) => p.instanceID.equals(rps.instanceID)) === i;
+            return ropascis.findIndex((p) => p.roPaSciID.equals(rps.roPaSciID)) === i;
         });
     }
 
     async wipeRPS() {
-        const ropasci = {wipe: true};
+        const ropasci = new RoPaSciList({wipe: true});
         await Promise.all(this.list.map(async (addr) => {
-            const socket = new WebSocket(addr.getWebSocketAddress(), PersonhoodRPC.serviceID);
-            // let resp = await socket.send("RoPaSciList", "RoPaSciListResponse", ropasci);
+            const socket = new WebSocketConnection(addr.getWebSocketAddress(), PersonhoodRPC.serviceID);
+            return socket.send(ropasci, RoPaSciListResponse);
```

Comment:
`return` and `await` would give the same results (as the return values is not read); IMHO, `await` is more readable then `return` in this case.

Reply:
Personal opinion: I prefer `return` here, because it shows that this is the end of the processing, and that no further processing will be done. If I read `await`, I think that there will be other lines.

Here it's a real corner-case, because it's also the end of the callback-method. Anyway, `resolve conversation` or reply with `change it anyway`.

Reply:
@ineiti-bot `change it anyway`

Reply:
one day the bots will revolt!